aggregators/codec.go

```diff
@@ -151,24 +154,22 @@ func (k *serviceInstanceAggregationKey) FromProto(pb *aggregationpb.ServiceInsta
 // ToProto converts ServiceInstanceMetrics to its protobuf representation.
 func (m *serviceInstanceMetrics) ToProto() *aggregationpb.ServiceInstanceMetrics {
 	pb := aggregationpb.ServiceInstanceMetricsFromVTPool()
-	if len(m.TransactionGroups) > cap(pb.TransactionMetrics) {
-		pb.TransactionMetrics = make([]*aggregationpb.KeyedTransactionMetrics, 0, len(m.TransactionGroups))
-	}
+
+	pb.TransactionMetrics = slices.Grow(pb.TransactionMetrics, len(m.TransactionGroups))
 	for _, m := range m.TransactionGroups {
 		pb.TransactionMetrics = append(pb.TransactionMetrics, m)
 	}
-	if len(m.ServiceTransactionGroups) > cap(pb.ServiceTransactionMetrics) {
-		pb.ServiceTransactionMetrics = make([]*aggregationpb.KeyedServiceTransactionMetrics, 0, len(m.ServiceTransactionGroups))
-	}
+
+	pb.ServiceTransactionMetrics = slices.Grow(pb.ServiceTransactionMetrics, len(m.ServiceTransactionGroups))
 	for _, m := range m.ServiceTransactionGroups {
 		pb.ServiceTransactionMetrics = append(pb.ServiceTransactionMetrics, m)
 	}
-	if len(m.SpanGroups) > cap(pb.SpanMetrics) {
-		pb.SpanMetrics = make([]*aggregationpb.KeyedSpanMetrics, 0, len(m.SpanGroups))
-	}
+
+	pb.SpanMetrics = slices.Grow(pb.SpanMetrics, len(m.SpanGroups))
 	for _, m := range m.SpanGroups {
 		pb.SpanMetrics = append(pb.SpanMetrics, m)
 	}
+
```

Comment:
I'm studying this piece of code here and trying to understand why we don't CloneVT here and what happens to the builder when all of this is returned to the VTPool. Will we be sharing memory between VTPool and the metric builder on reuse?

Reply:
ToProto only works on structures obtained from VT Unmarshal